homeassistant/components/fastdotcom/__init__.py

```diff
@@ -40,38 +35,49 @@
 )
 
 
-async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+async def async_setup_platform(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the Fast.com component. (deprecated)."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up the Fast.com component."""
-    conf = config[DOMAIN]
     data = hass.data[DOMAIN] = SpeedtestData(hass)
 
-    if not conf[CONF_MANUAL]:
-        async_track_time_interval(hass, data.update, conf[CONF_SCAN_INTERVAL])
+    async_track_time_interval(hass, data.update, timedelta(hours=DEFAULT_INTERVAL))
 
-    def update(service_call: ServiceCall | None = None) -> None:
+    async def update(service_call: ServiceCall | None = None) -> None:
         """Service call to manually update the data."""
-        data.update()
+        await data.update()
 
     hass.services.async_register(DOMAIN, "speedtest", update)
 
-    hass.async_create_task(
-        async_load_platform(hass, Platform.SENSOR, DOMAIN, {}, config)
+    hass.data.setdefault(DOMAIN, {})
+    await hass.config_entries.async_forward_entry_setups(
+        entry,
+        PLATFORMS,
     )
 
     return True
 
 
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload Fast.com config entry."""
+    return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+
+
 class SpeedtestData:
```

Comment:
In a follow-up PR I think it would be good to make this a proper `DataUpdaterCoordinator`

Reply:
Indeed, that is what my first PR had included. :)